### PR TITLE
optimization in isBoolean

### PR DESCRIPTION
**Pull request by zaheerehmad**
typeof on value and then comparison with string is slower then comparing value directly with boolean ie value === true. It will optimized the performance.

### CODE REVIEW

File: src/Angular.js [hunk 1]
```diff
@@ -580,7 +579,7 @@ function isBlob(obj) {
 
 
 function isBoolean(value) {
-  return typeof value === 'boolean';
+  return value === true;
```
Comment (caitp) [inline src/Angular.js]: this is semantically different from what was happening before.

**Comment by caitp** [comment]
I'd be interested in seeing profiling of an application which shows this is worth fixing in this way, because right now this does not lgtm

**Comment by lgalfaso** [comment]
This PR changes the semantics of the function, without fixing this and some numbers that show that it makes a difference, then this is not good for landing. Closing, but if you provide this information, feel free to reopen the PR

File: src/Angular.js (hunk 1 above)
Comment (Burgov) [inline src/Angular.js]: shouldn't it be `value === true || value === false`?

Comment (zaheer-ahmed) [inline src/Angular.js]: yes it should be. thanks

**Pull request closed without merging**